packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -228,6 +228,50 @@ class ScrollPhysics {
     return parent.applyBoundaryConditions(position, value);
   }
 
+  /// Describes what the scroll position should be given new viewport dimensions.
+  ///
+  /// This is called by [ScrollPosition.applyContentDimensions].
+  ///
+  /// The arguments consist of the current scroll metrics, including the
+  /// position and minimum and maximum scroll extents; a flag indicating if the
+  /// current [ScrollActivity] considers that the user is actively scrolling
+  /// (see [ScrollActivity.isScrolling]); and the current velocity of the scroll
+  /// position, if it is being driven by the scroll activity (this is 0.0 during
+  /// a user gesture) (see [ScrollActivity.velocity]).
+  ///
+  /// If the returned value does not exactly match the current scroll offset
+  /// given by the `position` argument (see [ScrollMetrics.pixels]), then the
+  /// [ScrollPosition] will call [ScrollPosition.correctPixels] to update the
+  /// new value, and layout will be re-run. This is expensive. The new value is
+  /// subject to further manipulation by [applyBoundaryConditions].
+  ///
+  /// If the returned value _does_ match the current scroll offset exactly, then
+  /// [ScrollPosition.applyNewDimensions] will be called next. In that case,
+  /// [applyBoundaryConditions] is not applied to the return value.
+  ///
+  /// The given `position` is only valid during this method call. Do not keep a
+  /// reference to it to use later, as the values may update, may not update, or
+  /// may update to reflect an entirely unrelated scrollable.
+  ///
+  /// ## Examples
+  ///
+  /// The default implementation returns [ScrollMetrics.pixels], which indicates
+  /// that the current scroll offset is acceptable.
+  ///
+  /// [ClampingScrollPhysics] returns the current scroll offset
+  /// ([ScrollMetrics.pixels]) clamped to the minimum and maximum scroll extents
+  /// ([ScrollMetrics.minScrollExtent] and [ScrollMetrics.maxScrollExtent]),
+  /// essentially forcing the scroll position to never go out-of-range even if
+  /// the contents of the list change suddenly.
+  double adjustPositionForNewDimensions(ScrollMetrics position, {
+    @required bool isScrolling,
+    @required double velocity,
+  }) {
+    return (parent == null)
+      ? position.pixels
+      : parent.adjustPositionForNewDimensions(position, isScrolling: isScrolling, velocity: velocity);
```

Comment:
we should use the style that is used in the other functions here, it's more readable than ?: imho (or rather, it more accurately represents the prototype inheritance that is being implemented -- it's not like we're considering two different alternatives, rather we're returning the default if there's no parent, and otherwise we're falling back on the parent).

Reply:
It sounds like you attach semantics to the ternary operator that aren't necessarily associated with it.  Usage of the ternary operator can help with 2 things:
1) Reduce mutations:
```
int value;
if (isTrue) {
  value = 1;
} else {
  value = 0;
}
```
2) Reduce return statements in a procedure:
```
int foo() {
  if (isTrue) {
    return 1;
  }
  return 0;
}
```

Both mutations and multiple exits on a procedure are generally regarded as things to be reduced to clean up code, so the ternary operator can help us there.

Reply:
mostly I'm just eager to keep the style in this file consistent.

Reply:
done